### PR TITLE
Add a post-deployment Ansible provisioner for the BOD bastion

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request adds a post-deployment Ansible provisioner for the BOD bastion.

## 💭 Motivation and context ##

This ensures that all of our EC2 instances are configured with an Ansible provisioner for post-deployment configuration needs. Although there is nothing specifically needed for the BOD bastion (the `groups` Ansible role will be run against it now) it ensures that the scaffolding exists for any future needs.

## 🧪 Testing ##

Automated tests pass. I was able to successfully deploy a new BOD bastion in my test environment and had no issues with functionality.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

File: terraform/bod_bastion_ec2.tf [hunk 1]
```diff
@@ -44,3 +44,19 @@ resource "aws_instance" "bod_bastion" {
...
   )
 }
+
+# Provision the bastion EC2 instance via Ansible
+module "bod_bastion_ansible_provisioner" {
+  source = "github.com/cloudposse/terraform-null-ansible"
+
+  arguments = [
+    "--user=${var.remote_ssh_user}",
+    "--ssh-common-args='-o StrictHostKeyChecking=no'",
```
Comment (dav3r) [inline terraform/bod_bastion_ec2.tf:54]: Alphabetize args:
```suggestion
    "--ssh-common-args='-o StrictHostKeyChecking=no'",
    "--user=${var.remote_ssh_user}",
```

**Review by dav3r — APPROVED** [review]
👍 
I noted one tiny thang.

File: terraform/bod_bastion_ec2.tf (hunk 1 above)
Comment (mcdonnnj) [reply terraform/bod_bastion_ec2.tf:54]: I simply copied the CyHy bastion's block and made necessary changes so if you are amenable I would like to do this in a separate PR that alphabetizes the values feeding all of the Ansible provisioner modules (both the SSH arguments and the environment variables where applicable).

Comment (dav3r) [reply terraform/bod_bastion_ec2.tf:54]: Sure, that's fine by me. 👍

**Review by jsf9k — APPROVED** [review]

**Pull request merged**